pandas/core/indexes/base.py

```diff
@@ -2320,6 +2320,11 @@ def symmetric_difference(self, other, result_name=None):
         except TypeError:
             pass
 
+        # On equal MultiIndexes the difference is empty. Therefore an empty
+        # MultiIndex is returned GH13490
+        if self.nlevels > 1 and len(the_diff) == 0:
+            return type(self)([[] for _ in range(self.nlevels)],
```

Comment:
> this change should be done in _shallow_copy_with_infer, you need to construct a MI equiv of self[0:0]

This leaves around some levels unfortunately:

```python
In [16]: idx = pd.MultiIndex.from_product([['a', 'b'], ['A', 'B']])

In [17]: idx[0:0]
Out[17]:
MultiIndex(levels=[['a', 'b'], ['A', 'B']],
           labels=[[], []])

In [18]: idx.difference(idx) | idx.difference(idx)
Out[18]:
MultiIndex(levels=[[], []],
           labels=[[], []])

```

I believe we want `Out[18]`